src/GovUk.Education.ExploreEducationStatistics.Admin/Areas/Identity/Pages/Account/ExternalLogin.cshtml.cs

```diff
@@ -68,15 +68,15 @@ public IActionResult OnPost(string provider, string returnUrl = null)
             return new ChallengeResult(provider, properties);
         }
 
-        public async Task<IActionResult> OnGetCallbackAsync(string providedReturnUrl = null, string remoteError = null)
+        public async Task<IActionResult> OnGetCallbackAsync(string returnUrl = null, string remoteError = null)
         {
-            var returnUrl = providedReturnUrl ?? Url.Content("~/");
+            var returnUrlOrFallback = returnUrl ?? Url.Content("~/");
```

Comment:
Not sure how this fixes the problem. It looks like we've just renamed some variables?

Reply:
It's the renaming of the argument name to `returnUrl` that fixed it. Duncan noticed that newer version of IdentityServer, the `OnGetCallbackAsync` method's argument name wasn't `providedReturnUrl`, so we speculatively changed it to `returnUrl` and then redirection worked. Dunc suggested IdentityServer uses reflection or something.

Reply:
Wow that's just mad. Can you leave a comment on the method saying that the parameter names are important?

Reply:
Done. I've also updated the PR description.